src/platform/packages/shared/response-ops/alerts-table/hooks/use_bulk_actions.ts

```diff
@@ -313,6 +320,98 @@ export const useBulkTagsActions = ({ refresh, clearSelection }: UseBulkTagsActio
   return { tagsAction };
 };
 
+const groupAlertsByRule = (alerts: TimelineItem[]) => {
+  const ruleMap = new Map<string, string[]>();
+
+  for (const alert of alerts) {
+    const ruleId = alert.data.find((d) => d.field === ALERT_RULE_UUID)?.value?.[0];
+    const alertInstanceId = alert.data.find((d) => d.field === ALERT_INSTANCE_ID)?.value?.[0];
+
+    if (ruleId && alertInstanceId) {
+      const existing = ruleMap.get(ruleId) || [];
+      existing.push(alertInstanceId);
+      ruleMap.set(ruleId, existing);
+    }
+  }
+
+  return Array.from(ruleMap.entries()).map(([ruleId, alertInstanceIds]) => ({
+    rule_id: ruleId,
+    alert_instance_ids: alertInstanceIds,
+  }));
+};
+
+export const useBulkMuteActions = ({
+  setIsBulkActionsLoading,
+  refresh,
+  clearSelection,
+  http,
+  notifications,
+}: UseBulkMuteActionsProps) => {
+  const onSuccess = useCallback(() => {
+    refresh();
+    clearSelection();
+  }, [clearSelection, refresh]);
+
+  const { mutateAsync: bulkMute } = useBulkMuteAlerts({ http, notifications });
+  const { mutateAsync: bulkUnmute } = useBulkUnmuteAlerts({ http, notifications });
+
+  const onMuteClick = useCallback(
+    async (selectedAlerts?: TimelineItem[]) => {
+      if (!selectedAlerts) return;
+      const rules = groupAlertsByRule(selectedAlerts);
+      if (rules.length === 0) return;
+
+      try {
+        setIsBulkActionsLoading(true);
```

Comment:
I was thinking of a different approach using React Queries callbacks. For example:

```
// will run before the mutation
const onMutate = () => setIsBulkActionsLoading(true)

// will run after the mutation for error or success
const onSettled = () => setIsBulkActionsLoading(false)

useBulkMuteAlerts({ http, notifications, onSuccess, onMutate, onSettled })
```

Feel free to ignore this suggestion, as it is the pattern we already follow everywhere. I just wanted to mention an alternative 🙂.